include/cetl/any.hpp

```diff
@@ -0,0 +1,344 @@
+/// @file
+/// Defines the C++17 `std::any` type and several related entities.
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef CETL_ANY_HPP_INCLUDED
+#define CETL_ANY_HPP_INCLUDED
+
+#include "pf17/cetlpf.hpp"
+#include "pf17/utility.hpp"
+#include "pf17/attribute.hpp"
+
+#include <algorithm>
+
+namespace cetl
+{
+
+// Forward declarations
+template <std::size_t Footprint, bool Copyable, bool Movable>
+class any;
+
+namespace detail
+{
+
+// Move policy.
+template <std::size_t Footprint, bool Copyable, bool Movable>
+struct base_move;
+
+// Copy policy.
+template <std::size_t Footprint, bool Copyable>
+struct base_copy;
+
+template <std::size_t Footprint, std::size_t Align = sizeof(std::max_align_t)>
```

Comment:
The alignment should be also configurable via the list of template parameters to any:

```c++
template <std::size_t Footprint,
          bool Copyable = true,
          bool Movable = Copyable,
          std::size_t Align = alignof(std::max_align_t)>
```

I neglected to mentioned this explicitly earlier. Also note that `alignof` should be used instead of `sizeof`.

Reply:
`Align` configurable will be ready (and tested) after next pr (already ready in my separate private branch).